state/protocol/inmem/epoch.go

```diff
@@ -23,30 +23,32 @@ func (eq Epochs) Previous() protocol.Epoch {
 	if eq.entry.PreviousEpoch == nil {
 		return invalid.NewEpoch(protocol.ErrNoPreviousEpoch)
 	}
-	return NewCommittedEpoch(eq.entry.PreviousEpochSetup, eq.entry.PreviousEpochCommit)
+	return NewCommittedEpoch(eq.entry.PreviousEpochSetup, eq.entry.PreviousEpoch.EpochExtensions, eq.entry.PreviousEpochCommit)
 }
 
 func (eq Epochs) Current() protocol.Epoch {
-	return NewCommittedEpoch(eq.entry.CurrentEpochSetup, eq.entry.CurrentEpochCommit)
+	return NewCommittedEpoch(eq.entry.CurrentEpochSetup, eq.entry.CurrentEpoch.EpochExtensions, eq.entry.CurrentEpochCommit)
 }
 
 func (eq Epochs) Next() protocol.Epoch {
 	switch eq.entry.EpochPhase() {
 	case flow.EpochPhaseStaking, flow.EpochPhaseFallback:
 		return invalid.NewEpoch(protocol.ErrNextEpochNotSetup)
 	case flow.EpochPhaseSetup:
-		return NewSetupEpoch(eq.entry.NextEpochSetup)
+		return NewSetupEpoch(eq.entry.NextEpochSetup, eq.entry.NextEpoch.EpochExtensions)
```

Comment:
In my opinion, this should yield an error `ErrNextEpochNotCommitted`
```golang
return invalid.NewEpoch(protocol.ErrNextEpochNotCommitted)
```
Though, this would require an update of the [`EpochQuery`  API](https://github.com/onflow/flow-go/blob/430a00cad5835f5cb132882df8da8954fb41bd72/state/protocol/epoch.go#L9) ... so maybe add a todo to the issue is mentioned in my previous comment.

Reply:
See https://github.com/onflow/flow-go/pull/6154#discussion_r1666853621